jaeger-core/src/main/java/io/jaegertracing/senders/SenderFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2018, The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.jaegertracing.senders;
+
+import io.jaegertracing.Configuration;
+
+/**
+ * Represents a class that knows how to select and build the appropriate {@link Sender} based on the given
+ * {@link Configuration.SenderConfiguration}
+ */
+public interface SenderFactory {
+  /**
+   * Builds and/or selects the appropriate sender based on the given {@link Configuration.SenderConfiguration}
+   * @param senderConfiguration the sender configuration
+   * @return an appropriate sender based on the configuration, or {@link NoopSender}.
```

Comment:
Not really -- I expect concrete `SenderFactory` implementations to *never* return null. The contract is: either a real `Sender`, or `NoopSender`.

Reply:
What can't we return null? It's invoked only in `SenderFactory` which takes care about null/noop.

Reply:
This is a contract between the implementation and the consumer, meaning that this documentation is a reference to both sides. By stating that it might return NoopSender, we are ruling out a null object, which makes it far easier for consuming code to not break (no null checks, no null pointer exceptions).

Reply:
I still think the Noop here is just redundant. This API does not need it and it's just creating a dependency on specific sender impl